File: packages/registro/src/usecases/ObterRegistros.ts
import { CasoDeUso } from "common";
import { Registro } from "../model";
import { BackendRegistro } from "../provider";

export class ObterRegistros implements CasoDeUso<void, Registro[]> {
  public constructor(private backend: BackendRegistro) {}
  async executar(dto: void): Promise<Registro[]> {
    const response = await this.backend.obterRegistros();
    if (!response) return [];

    return response;
  }
}

File: apps/backend-springboot/src/main/java/br/com/dinheironapalmadamao/backend_springboot/core/registro/usecases/ObterRegistros.java
package br.com.dinheironapalmadamao.backend_springboot.core.registro.usecases;

import java.util.concurrent.CompletableFuture;

import br.com.dinheironapalmadamao.backend_springboot.core.common.CasoDeUso;
import br.com.dinheironapalmadamao.backend_springboot.core.registro.model.Registro;
import br.com.dinheironapalmadamao.backend_springboot.core.registro.provider.BackendRegistro;

public class ObterRegistros implements CasoDeUso<Void, Registro[]> {
  private BackendRegistro backend;

  public ObterRegistros(BackendRegistro backend) {
    this.backend = backend;
  }

  @Override
  public CompletableFuture<Registro[]> excutar(Void dto) {
    return CompletableFuture.supplyAsync(() -> {
      Registro[] registros = this.backend.obterRegistros();

      // Se registros for nulo ou vazio, retorne um array vazio
      if (registros == null || registros.length == 0) {
        return new Registro[0];
      }

      return registros;
    });

  }

}
